### DISCUSSION TITLE
Scratch docker container

### DISCUSSION THREAD

**Discussion by aslafy-z** [jart/cosmopolitan#1333] [Q&A]
Hello,
I would like to bundle some apps into a scratch container for it to work with multi-arch without having to publish multiple images.

I've tried 

```dockerfile
FROM scratch
ADD --chmod=755 https://cosmo.zip/pub/cosmos/v/3.9.2/bin/dash /bin/sh
ADD --chmod=755 https://cosmo.zip/pub/cosmos/v/3.9.2/bin/cp.ape /bin/cp
ENTRYPOINT ["/bin/sh"]
```

But it fails with
```
exec /bin/sh: exec format error
```

Any idea if it is possible?

**Comment by franalbani** [comment]
I guess those files are APE, so you need to wrap them with `ape` or use the `binfmt_misc` trick.

Try adding this:

```Dockerfile
ADD --chmod=755 https://cosmo.zip/pub/cosmos/v/3.9.2/bin/ape-x86_64.elf /bin/ape
```

And then wrap `sh` with it. But already we can see that it is for `x86_64` so it won't achieve what you want.

I'm curious why you need docker just to bundle some binaries?

Is your idea that `docker run thisimage` will open a shell+utils in every platform?

Could we achieve the same using the zipos feature of APE binaries?
